crates/net/network-api/src/lib.rs

```diff
@@ -12,19 +12,38 @@
 use async_trait::async_trait;
 use reth_primitives::{NodeRecord, H256, U256};
 use serde::{Deserialize, Serialize};
-use std::{error::Error, net::SocketAddr};
+use std::net::SocketAddr;
+use thiserror::Error;
+use tokio::sync::{mpsc, oneshot};
+
+/// Network Errors
+#[allow(missing_docs)]
+#[derive(Error, Debug, Clone, PartialEq, Eq)]
+pub enum NetworkError {
+    #[error("Sender has been dropped")]
+    ChannelClosed,
+}
+
+impl<T> From<mpsc::error::SendError<T>> for NetworkError {
+    fn from(_: mpsc::error::SendError<T>) -> Self {
+        NetworkError::ChannelClosed
+    }
+}
+
+impl From<oneshot::error::RecvError> for NetworkError {
+    fn from(_: oneshot::error::RecvError) -> Self {
+        NetworkError::ChannelClosed
+    }
+}
```

Comment:
one last nit, sorry for being a bit pedantic, but would appreciate if you could move this to error.rs and  `pub use` from lib.rs

Reply:
Sure. Cleaner that way.